src/storage/mvcc/reader.rs

```diff
@@ -401,9 +407,74 @@ impl MvccReader {
                     },
                 }
             };
-            if let Some(v) = self.get(&key, ts)? {
+
+            if let Some(v) = self.reverse_get_impl(&key, ts)? {
                 return Ok(Some((key, v)));
             }
+            if !self.write_cursor.as_mut().unwrap().valid() {
+                write_valid = false;
+            }
+        }
+    }
+
+    fn reverse_get_impl(&mut self, user_key: &Key, ts: u64) -> Result<Option<Value>> {
+        assert!(self.lock_cursor.is_some());
+        assert!(self.write_cursor.is_some());
+
+        // Check lock.
+        match self.isolation_level {
```

Comment:
can we eliminate a `load_lock` for `get()` from `seek()` as well?

Reply:
Actually inside the Cursor's near_seek, the same thing has been done. But I think we still can make it simple and clear when we refine this mode.